src/cargo/util/semver_ext.rs

```diff
@@ -117,9 +117,9 @@ impl OptVersionReq {
     /// and we're not sure if this part of the functionality should be implemented in semver or cargo.
     pub fn matches_prerelease(&self, version: &Version) -> bool {
         if version.is_prerelease() {
-            let mut version = version.clone();
-            version.pre = semver::Prerelease::EMPTY;
-            return self.matches(&version);
+            let mut version_cleaned = version.clone();
+            version_cleaned.pre = semver::Prerelease::EMPTY;
+            return self.matches(&version) || self.matches(&version_cleaned);
```

Comment:
This needs more looking into. A test is failing. I'll also move this and related changes to a separate commit.

Reply:
Two commits inserted at the beginning of the PR.